### ISSUE TITLE
Document Can't Create Non-Root User Connection to CockroachDB in Insecure Mode

### ISSUE THREAD

**Issue by stephenandary** [ory/docs#1525]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

For getting started documentation, please highlight that one is unable to connect to cockrachdb from Kratos with non-root user if crdb is started in insecure mode because can't set password in crdb during service start.

Thank you Adam Strawson in Slack community for figuring this out. 

Thread here: https://ory-community.slack.com/archives/C02PSRUGUAV/p1693403474573559

TLDR: 
Per Adam: 
So if you want to use --insecure  you can't use COCKROACH_USER etc to provision the user. You either have to create the user yourself, or using docker-entrypoint-initdb.d to create them after startup.
If you want to use COCKROACH_USER  then you'll need to setup secure mode properly (edited) 

### Reproducing the bug

1. Try to connect to Cockroachdb with non-root user in insecure mode from a kratos service

example docker compose in shared slack thread. 

### Version

1.0.0

### On which operating system are you observing this issue?

None

### In which environment are you deploying?

None
